src/Sylius/Bundle/UserBundle/spec/Form/EventListener/GuestCustomerFormSubscriberSpec.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Bundle\UserBundle\Form\EventListener;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\UserInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\User\Context\CustomerContextInterface;
+use Sylius\Component\User\Model\CustomerInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class GuestCustomerFormSubscriberSpec extends ObjectBehavior
+{
+    function let(RepositoryInterface $customerRepository, FactoryInterface $customerFactory, CustomerContextInterface $customerContext)
+    {
+        $this->beConstructedWith($customerRepository, $customerFactory, $customerContext);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Bundle\UserBundle\Form\EventListener\GuestCustomerFormSubscriber');
+    }
+
+    function it_implements_event_subscriber_interface()
+    {
+        $this->shouldImplement('Symfony\Component\EventDispatcher\EventSubscriberInterface');
+    }
+
+    function it_sets_currently_logged_user_as_form_data(
+        $customerContext,
+        CustomerInterface $customer,
+        FormEvent $event,
+        FormInterface $form
+    ) {
+        $event->getData()->willReturn(array('email' => null))->shouldBeCalled();
+        $event->getForm()->willReturn($form)->shouldBeCalled();
+
+        $customerContext->getCustomer()->willReturn($customer)->shouldBeCalled();
+
+        $form->remove('email')->shouldBeCalled();
+        $form->setData($customer)->shouldBeCalled();
+
+        $this->preSubmit($event);
+    }
+
+    function it_sets_new_customer_with_passed_email_as_form_data_if_customer_with_such_email_does_not_exist(
+        $customerContext,
+        $customerFactory,
+        $customerRepository,
+        CustomerInterface $customer,
+        FormEvent $event,
+        FormInterface $form
+    ) {
+        $event->getData()->willReturn(array('email' => 'john.doe@example.com'))->shouldBeCalled();
```

Comment:
`shouldBeCalled()` should be removed as you already use `willReturn()`, same below.